gitpod-io/gitpod
# [gitpod-cli] add command analytics

## Description
This PR adds analytics to the `gp` CLI. This will help us understand which commands are most and least useful.

All commands updated

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/10349

## How to test

1. start a workspace in this preview environment
2. use [segment](https://app.segment.com/gitpod/sources/staging_trusted/debugger) to check event.
3. all commands should send track event, except `send-analytics` itself, if the command success, `outcome` should be `success`, if the command failed, the `outcome` should be `system_error` or `user_error` depending on the error.
4. we might not able to trace `gp stop` command, because at that time, the workspace is going to stop.

- [ ] ~~credential-helper~~
- [x] env
- [ ] ~~git-token-validator~~
- [ ] ~~git-track-command~~
- [x] info
- [x] init
- [x] ports await
- [x] ports expose - skip
- [x] ports list
- [x] ports visibility
- [x] rebuild
- [x] snapshot
- [x] stop
- [x] sync await
- [x] sync done
- [x] tasks attach
- [x] tasks list
- [x] tasks stop
- [x] timeout extend
- [x] timeout set
- [x] timeout show
- [x] top
- [x] url

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
